mypy/typeanal.py

```diff
@@ -860,17 +885,22 @@ def anal_type(self, t: Type, nested: bool = True) -> Type:
             if nested:
                 self.nesting_level -= 1
 
-    def anal_var_defs(self, var_defs: List[TypeVarDef]) -> List[TypeVarDef]:
-        a = []  # type: List[TypeVarDef]
-        for vd in var_defs:
-            a.append(TypeVarDef(vd.name,
-                                vd.fullname,
-                                vd.id.raw_id,
-                                self.anal_array(vd.values),
-                                vd.upper_bound.accept(self),
-                                vd.variance,
-                                vd.line))
-        return a
+    def anal_var_def(self, var_def: TypeVarLikeDef) -> TypeVarLikeDef:
+        if isinstance(var_def, TypeVarDef):
+            return TypeVarDef(
+                var_def.name,
+                var_def.fullname,
+                var_def.id.raw_id,
+                self.anal_array(var_def.values),
+                var_def.upper_bound.accept(self),
+                var_def.variance,
+                var_def.line
+            )
+        else:
+            return var_def
```

Comment:
Should we construct a new instance and set the full name, at least?

Reply:
In general, TypeAnalyser visit methods seem pretty happy to return their input, so not sure I see the point (unless we were to add upper_bound to ParamSpecDef or TypeVarLikeDef, in which case we'd need to visit it and reconstruct).

Reply:
I was confused -- no change is needed here.